.github/workflows/ci.yaml

```diff
@@ -40,6 +40,12 @@ jobs:
           key: ${{ matrix.os }}-cargo-target-${{ matrix.rust_version }}-${{ hashFiles('**/Cargo.lock') }}
           restore-keys: |
             ${{ matrix.os }}-cargo-target-${{ matrix.rust_version }}-
+      - name: Add Gosub crate registry
+        run: |
+          cat <<EOF > ~/.cargo/config.toml
+          [registries]
+          gosub = { index = "sparse+https://registry.gosub.io/" }
+          EOF
```

Comment:
this should be just in `gosub-engine/.cargo/config.toml` not in `~/.cargo/config.toml` so when you clone the engine you don't need to modify something in your home folder